src/components/CreateApplication/CreateApplicationPage.tsx

```diff
@@ -27,6 +34,16 @@ export const CreateApplicationPage = () => {
     setValues({});
   };
 
+  React.useEffect(() => {
+    const existingApplication = queryParams.get('application');
+    if (existingApplication) {
+      setValues((prevVal) => ({ ...prevVal, ...initialValues, existingApplication }));
+      handleNext();
+      queryParams.delete('application');
+      history.replace({ search: queryParams.toString() });
+    }
+  }, [handleNext, history, initialValues, queryParams, setValues]);
```

Comment:
Yes, we do this since the user comes from component list view under a specific application and user expects to add the component to an existing application.